### ISSUE TITLE
Wrong scroll position after orientation change in layout picker

### ISSUE THREAD

**Issue by mkevins** [wordpress-mobile/WordPress-iOS#16178]
**Note:** This bug is also present in the Site Creation flow in the Site Design picker

### Expected behavior
Scroll position should stay at or near the layout category that was visible prior to the orientation change.

### Actual behavior
The scroll position is far from where it was prior to the orientation change.

### Steps to reproduce the behavior
* View the page layout picker
* Scroll down a bit
* Rotate the device

##### Tested on iPhon11, iOS 14.4.1, WPiOS 16.9

**Comment by antonis** [comment]
I tried various ways of handling the scroll position manually in [`FilterableCategoriesViewController`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Gutenberg/Layout%20Picker/FilterableCategoriesViewController.swift) that is used in both the Site Creation flow and the Site Design picker (e.g. saving the `indexPathsForVisibleRows` or `contentOffset` on rotation). The sticky header combined with the horizontal scrolling make the implementation too complicated and I didn't manage to come up with a good solution.
Given that this screen is going to be replaced with a full screen design selection (at least for the Site Design picker, ref: `pc8HXX-6Z-p2`) and the fact that the current behaviour on iPad (where landscape mode is more common) is decent I won't spend more time on this for now.

<details>
<summary>iPad recording</summary>

https://user-images.githubusercontent.com/304044/154514341-f8949c10-607d-4478-9c01-381097c455bc.mp4
</details>

**Comment by mkevins** [comment]
> Given that this screen is going to be replaced with a full screen design selection (at least for the Site Design picker, ref: `pc8HXX-6Z-p2`) and the fact that the current behaviour on iPad (where landscape mode is more common) is decent I won't spend more time on this for now.

I agree with this, that since there doesn't seem to be a "low hanging fruit" solution to this, investing a lot of effort on it wouldn't be worth it when we will be replacing these screens soon. Thanks for exploring some potential solutions to this Antonis. :+1:

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
